File: frontend/src/app/models/payment.ts
export class Payment {
  public description: string;
  public paymentDate: Date;
  public paymentType: number;
  public paymentAmount: number;

  constructor(description: string, paymentDate: Date,  paymentType: number, paymentAmount: number) {
    this.description = description;
    this.paymentDate = paymentDate;
    this.paymentType = paymentType;
    this.paymentAmount = paymentAmount;
  }
}

File: backend/Payment/Domain/Models/Payment.cs
using Domain.Enums;

namespace Domain.Models;

public class Payment : Entity
{
    public Payment(string desciption,
                   DateTime paymentDate,
                   PaymentType paymentType,
                   decimal paymentAmount)
    {
        Description = desciption;
        PaymentDate = paymentDate;
        PaymentType = paymentType;
        PaymentAmount = paymentAmount;
    }

    public string Description { get; private set; }
    public DateTime PaymentDate { get; private set; }
    public PaymentType PaymentType { get; private set; }
    public decimal PaymentAmount { get; private set; }
}
